eskip/eskip.go

```diff
@@ -20,6 +20,20 @@ var (
 	duplicateMethodPredicateError   = errors.New("duplicate method predicate")
 )
 
+type DefaultFilters struct {
+	Prepend []*Filter
+	Append  []*Filter
+}
+
+func (df *DefaultFilters) Do(r []*Route) []*Route {
+	for i := range r {
```

Comment:
we should clone the routes objects. The reason is that a DataClient may cache parsed routes, and this logic changes the Filters field on those routes, too. This way we may end up prepending and appending to the same slice of filters.

Reply:
done